src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -3504,14 +3504,15 @@ public String toPlainString() {
             }
             int trailingZeros = checkScaleNonZero((-(long)scale));
             StringBuilder buf;
-            if(intCompact!=INFLATED) {
-                buf = new StringBuilder(20+trailingZeros);
-                buf.append(intCompact);
-            } else {
-                String str = intVal.toString();
-                buf = new StringBuilder(str.length()+trailingZeros);
-                buf.append(str);
+            String str = intCompact != INFLATED
+                ? Long.toString(intCompact)
+                : intVal.toString();
+            int len = str.length() + trailingZeros;
+            if (len < 0) {
+                throw new OutOfMemoryError("too large to fit in a String");
             }
+            buf = new StringBuilder(len);
+            buf.append(str);
             for (int i = 0; i < trailingZeros; i++) {
                 buf.append('0');
             }
```

Comment:
A bit more compact as:
```suggestion
            StringBuilder buf = new StringBuilder(len);
            buf.append(str);
            buf.repeat('0', trailingZeros);
```

Reply:
More compact and potentially faster.